File: week2/problemset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int words = 0;
    int letters = 0;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isspace(text[i]))
        {
            words++;
        }
        else if (text[i] == '?' || text[i] == '!' || text[i] == '.')
        {
            sentences++;
        }
    }
    words++;

    float L = 100.0 * letters / words;
    float S = 100.0 * sentences / words;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

File: week6/problemset/sentimental-readability/readability.py
import re
import math


def main():
    text = input("Text: ")

    words = len(text.split())
    letters = sum(c.isalpha() for c in text)
    sentences = len(re.split(r'[.!?]\s+', text))

    L = 100.0 * (letters / words)
    S = 100.0 * (sentences / words)

    index = 0.0588 * L - 0.296 * S - 15.8
    
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


main()
